examples/connect.rs

```diff
@@ -43,50 +39,50 @@
 extern crate unwrap;
 extern crate tokio_core;
 extern crate futures;
+extern crate future_utils;
 #[macro_use]
 extern crate serde_derive;
 extern crate serde;
 extern crate serde_json;
 #[macro_use]
 extern crate rand_derive;
 extern crate rand;
+extern crate clap;
+extern crate void;
 
 extern crate crust;
 
+mod utils;
 
-use crust::{ConfigFile, PubConnectionInfo, Service, Uid};
+use clap::App;
+use crust::{ConfigFile, PubConnectionInfo, Service};
+use future_utils::FutureExt;
 use futures::Stream;
-
-use futures::future::empty;
+use futures::future::{Future, empty};
 use rand::Rng;
-use std::{fmt, io};
-use std::path::PathBuf;
 use tokio_core::reactor::Core;
+use utils::{PeerId, read_line};
 
-// Some peer ID boilerplate.
-
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize, Rand)]
-struct PeerId(u64);
-
-impl Uid for PeerId {}
-
-impl fmt::Display for PeerId {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        let PeerId(ref id) = *self;
-        write!(f, "{:x}", id)
-    }
-}
 
 fn main() {
+    let _ = App::new("Crust basic connection example")
+        .about(
+            "Attempts to connect to remote peer given it's connection information.
+Start two instances of this example. Each instance generates and prints it's
+connection information to stdout in JSON format. You have to manually copy/paste
```

Comment:
In rust you can format a string by putting a `\` at the end of a line, causing all whitespace at the start of the next line to be ignored. eg.

```
    "hello\
  povilas \
          how are you\n\
 today?"
```
is the same as
```
"hellopovilas how are you
today?"